posts/Yoann/2023-06-02-hibernate-search-6-2-0-Beta2.adoc

```diff
@@ -0,0 +1,300 @@
+
+For example, one can enable highlighting on a full-text field like this:
+
+[source, JAVA, indent=0]
+----
+@Entity
+@Indexed
+public class Book {
+    @Id
+    private Integer id;
+
+    @FullTextField(analyzer = "english")
+    private String title;
+
+    @FullTextField(analyzer = "english", projectable = Projectable.YES)
+    private String description;
+}
+----
+
+And then retrieve highlights like this:
+
+[source, JAVA, indent=0]
+----
+SearchSession searchSession = /* ... */
+
+List<List<String>> result = searchSession.search( Book.class )
+        .select( f -> f.highlight( "description" ) )
+        .where( f -> f.match().field( "description" ).matching( "robot" ) )
+        .fetchHits( 20 );
+----
+
+Which can for example return the following highlights:
+
+[source, indent=0]
+----
+# Hit #0
+["A <em>robot</em> becomes self-aware."]
+# Hit #1
+["A <em>robot</em> helps investigate a murder on an extrasolar colony.", "On this planet, <em>robots</em> are used extensively."]
+# etc.
+----
+
+This is a simple example, but highligting can be configured extensively.
+See link:{hsearch-doc-url-prefix}#search-dsl-highlighting[this section of the reference documentation]
+for more information.
+
+[[indexing-plan-filter]]
+=== Indexing plan filters
+
+With link:{hsearch-jira-url-prefix}/HSEARCH-168[HSEARCH-168]/link:{hsearch-jira-url-prefix}/HSEARCH-1383[HSEARCH-1383],
+Hibernate Search now allow enabling/disabling indexing in indexing plans (which include in particular automatic indexing),
+both fully (for all types) or on a per-type basis.
+
+For example, the following will disable automatic indexing
+for future changes on all instances of the class `Company`,
+except for instances of its subclass `Customer`:
+
+[source, JAVA, indent=0]
+----
+SearchMapping searchMapping = /* ... */
+searchMapping.indexingPlanFilter(
+        ctx -> ctx.exclude( Company.class )
+                .include( Customer.class )
+);
+----
+
+Indexing plan filters can also be configured per-session
+(though limitations apply when using the link:{hsearch-doc-url-prefix}#coordination-outbox-polling[`outbox-polling` coordination strategy]):
+
+[source, JAVA, indent=0]
+----
+SearchSession searchSession = /* ... */
+searchSession.indexingPlanFilter(
+        ctx -> ctx.exclude( Company.class )
+                .include( Customer.class )
+);
+----
+
+See link:{hsearch-doc-url-prefix}#indexing-plan-filter[this section of the reference documentation]
+for more information.
+
+[[mapping-projection]]
+=== Mapping annotations for constructor parameters when using `@ProjectionConstructor`
+
+link:{hsearch-jira-url-prefix}/HSEARCH-4574[HSEARCH-4574] adds mapping annotations for constructor parameters when using `@ProjectionConstructor`.
+
+This means in particular you can now project to other things than just fields when using `@ProjectionConstructor`,
+for example the entity identifier:
```

Comment:
Maybe it's worth mentioning in this section that users wouldn't need to compile with `-parameters` now and use the annotations instead.

Reply:
Technically the examples always use the canonical constructor of a record, so people don't necessarily need `-parameters` (but of course it doesn't hurt).
Since that's the approach I've been trying to push for throughout the documentation, I'd rather not emphasize the need for `-parameters` too much here.